lib/auth0/client-factory.js

```diff
@@ -4,37 +4,25 @@
  * @module auth0/client-factory
  */
 
-const request = require('request');
+const memoize = require('lodash.memoize');
+const getAuth0Configuration = require('./configuration.js');
 
 /**
  * @function getClientIdByName
  * @param {String} clientName - The name of a Client.
  * @returns {String} The Id of the Client.
  */
 function getClientIdByName (clientName) {
-  return new Promise((resolve, reject) => {
-    request.get('https://auth.blinkm.io/auth0-client-factory.json', (error, status, body) => {
-      if (error) {
-        return reject(error);
-      }
+  return getAuth0Configuration().then(auth0Configuration => {
+    const clientId = auth0Configuration.clients[clientName];
+    if (!clientId) {
+      return Promise.reject(`Could not find Auth0 Client Id for Client: ${clientName}`);
```

Comment:
Just as we wouldn't `throw` a string literal, I'd recommend that we don't `Promise#reject()` a string literal:

- http://eslint.org/docs/rules/no-throw-literal

- https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

It's better to `throw` and `Promise#reject()` an actual `Error` object when possible, as it can provide additional information useful for debugging and error handling. By making this a convention, it also means our error handling code can assume an `Error` object instead of also having to check for a `String`.